404.php

```diff
@@ -6,11 +6,286 @@
+  },
+
+  preparePoints: function() {
+
+    // Clear the current points
+    this.points = [];
+
+    var width, height, i, j;
+
+    var colors = this.bgContextPixelData.data;
+
+    for( i = 0; i < this.canvas.height; i += this.density ) {
+
+      for ( j = 0; j < this.canvas.width; j += this.density ) {
+
+        var pixelPosition = ( j + i * this.bgContextPixelData.width ) * 4;
+
+        // Dont use whiteish pixels
+        if ( colors[pixelPosition] > 200 && (colors[pixelPosition + 1]) > 200 && (colors[pixelPosition + 2]) > 200 || colors[pixelPosition + 3] === 0 ) {
+          continue;
+        }
+
+        var color = 'rgba(' + colors[pixelPosition] + ',' + colors[pixelPosition + 1] + ',' + colors[pixelPosition + 2] + ',' + '1)';
+        this.points.push( { x: j, y: i, originalX: j, originalY: i, color: color } );
+
+      }
+    }
+  },
+
+  updatePoints: function() {
+
+    var i, currentPoint, theta, distance;
+
+    for (i = 0; i < this.points.length; i++ ){
+
+      currentPoint = this.points[i];
+
+      theta = Math.atan2( currentPoint.y - this.mouse.y, currentPoint.x - this.mouse.x);
+
+      if ( this.mouse.down ) {
+        distance = this.reactionSensitivity * 400 / Math.sqrt((this.mouse.x - currentPoint.x) * (this.mouse.x - currentPoint.x) +
+         (this.mouse.y - currentPoint.y) * (this.mouse.y - currentPoint.y));
+      } else {
+        distance = this.reactionSensitivity * 200 / Math.sqrt((this.mouse.x - currentPoint.x) * (this.mouse.x - currentPoint.x) +
+         (this.mouse.y - currentPoint.y) * (this.mouse.y - currentPoint.y));
+      }
+
+
+      currentPoint.x += Math.cos(theta) * distance + (currentPoint.originalX - currentPoint.x) * 0.05;
+      currentPoint.y += Math.sin(theta) * distance + (currentPoint.originalY - currentPoint.y) * 0.05;
+
+    }
+  },
+
+  drawLines: function() {
+
+    var i, j, currentPoint, otherPoint, distance, lineThickness;
+
+    for ( i = 0; i < this.points.length; i++ ) {
+
+      currentPoint = this.points[i];
+
+      // Draw the dot.
+      this.context.fillStyle = currentPoint.color;
+      this.context.strokeStyle = currentPoint.color;
+
+      for ( j = 0; j < this.points.length; j++ ){
+
+        // Distaqnce between two points.
+        otherPoint = this.points[j];
+
+        if ( otherPoint == currentPoint ) {
+          continue;
+        }
+
+        distance = Math.sqrt((otherPoint.x - currentPoint.x) * (otherPoint.x - currentPoint.x) +
+         (otherPoint.y - currentPoint.y) * (otherPoint.y - currentPoint.y));
```

Comment:
alternatively, since you're doing this more than once, i might make a helper as follows:
```js
computeDistance: function(diffX, diffY){
  return Math.sqrt( Math.pow(diffX, 2) - Math.pow(diffY, 2));
}
```

if you wanted, you could also make it a lamda:
```js
let distance = { diffX, diffY } => Math.sqrt( Math.pow(diffX, 2) - Math.pow(diffY,2) );
```